src/Integrator/BDHI/DoublyPeriodic/DPStokesSlab.cuh

```diff
@@ -96,6 +96,44 @@ namespace uammd{
 	return {particleVelocities, particleAngularVelocities};
       }
 
+      // compute average velocity in the x direction as a function of z
+      template<class PosIterator, class ForceIterator>
+      std::vector<double>
+      computeAverageVelocity(PosIterator pos, ForceIterator forces,
+			     int numberParticles, cudaStream_t st = 0){
+      	cudaDeviceSynchronize();
+      	System::log<System::DEBUG2>("[DPStokes] Computing displacements");
+      	auto gridData = ibm->spreadForces(pos, forces, numberParticles, st);
+      	auto gridForceCheb = fct->forwardTransform(gridData, st);
+      	FluidData<complex> fluid = solveBVPVelocity(gridForceCheb, st);
+      	if(mode != WallMode::none){
+      	  correction->correctSolution(fluid, gridForceCheb, st);
+      	}
+	int nx = this->grid.cellDim.x, ny = this->grid.cellDim.y, nz = this->grid.cellDim.z;
+	// std::cout << "printing average velocity" << std::endl;
```

Comment:
Remove

Reply:
The way I rewrote it, I still need:
int3 n = this->grid.cellDim;

Please see the updated file.

Reply:
Sorry, I meant the comment -.-

Reply:
That's done.